docs/prerequisites.md

```diff
@@ -6,7 +6,7 @@ AWS supports two naming conventions: [IP-based or resource-based naming](https:/
 
 When _IP-based naming_ is used, the nodes must be named after the instance followed by the regional domain name (`ip-xxx-xxx-xxx-xxx.ec2.<region>.internal`). If you have custom domain name set in the DHCP options, you must set `--hostname-override` on kube-proxy and kubelet to match the above-mentioned naming convention.
```

Comment:
Maybe revisit these docs in a separate PR, but this isn't accurate. The `--hostname-override` shouldn't always be set to match the `$REGION.compute.internal` domain (example here is also wrong btw), it should be set to whatever convention `PrivateDnsName` is using, depending the VPC's DNS attributes.

Reply:
That's a question I had, how does RBN on regular Ubuntu work currently where you get a node name of `i-01234567890`, which isn't the `PrivateDnsName`? Agree it can wait until we get a really good handle on documenting this carefully.

Reply:
That would work if your `configmap/aws-auth` had a mapping from the node role to `system:node:{{SessionName}}`, because the session name is the instance ID. With the standard `system:node:{{EC2PrivateDnsName}}`, it wouldn't work (FWICT)

Reply:
I think this is all about CCM supporting the distros own hostname convention and not what AWS may put in the metadata API or similar. There may be edge cases, although I cannot really think of one, but `--hostname-override` is already discouraged and it complicates node bootstrapping.

Reply:
I think it's more than just the hostname convention.  e.g. EKS creates an aws-auth configmap of:

```
apiVersion: v1
data:
  mapRoles: |
    - groups:
      - system:bootstrappers
      - system:nodes
      rolearn: arn:aws:iam::12345678901:role/role-name
      username: system:node:{{EC2PrivateDNSName}}
```
So the hostname must match the EC2PrivateDNSName using the default config.

Reply:
Using the name from the EC2 API is one of the things we want to explicitly avoid. So it sounds like this is something that may be changed on the EKS side to match what we are doing here.